trunk/source/Library-CHC/src/de/uni_freiburg/informatik/ultimate/lib/chc/EldaricaCliChcScript.java

```
@@ -0,0 +1,207 @@
+			final long defaultTimeout, final Path predicates) {
+		mServices = services;
+		mLogger = services.getLoggingService().getLogger(getClass());
+		mMgdScript = mgdScript;
+		mDefaultQueryTimeout = defaultTimeout;
+		mPredicates = predicates;
+	}
+
+	@Override
+	public Script getScript() {
+		return mMgdScript.getScript();
+	}
+
+	@Override
+	public LBool solve(final HcSymbolTable symbolTable, final List<HornClause> system) {
+		return solve(symbolTable, system, -1L);
+	}
+
+	@Override
+	public LBool solve(final HcSymbolTable symbolTable, final List<HornClause> system, final long timeout) {
+		// generate file with Horn clauses
+		final File tmpFile;
+		try {
+			tmpFile = File.createTempFile("eldarica_", ".smt2");
+			mLogger.info("Writing script to file " + tmpFile.getAbsolutePath());
+
+			final var dumperScript = new LoggingScript(tmpFile.getAbsolutePath(), true);
+			dumperScript.setLogic(Logics.HORN);
+
+			new ChcAsserter(mMgdScript, dumperScript, ADD_CLAUSE_NAMES, ADD_COMMENTS, DECLARE_FUNCTIONS)
+					.assertClauses(symbolTable, system);
+			dumperScript.checkSat();
+		} catch (final IOException e) {
+			throw new RuntimeException(e);
+		}
+
+		try {
+			// run eldarica on file
+			final var executor = new Executor(getCommand(tmpFile), mMgdScript.getScript(), mLogger, mServices,
+					"eldarica", null, null, null, determineTimeout(timeout));
+
+			if (mPredicates != null) {
+				// Skip the line in stdout that says:
+				// "---------------------------- Loading CEGAR hints -------------------------------"
+				executor.skipLine();
+			}
+
+			mLastResult = executor.parseCheckSatResult();
+			mLastModel = (mLastResult == LBool.SAT && mProduceModels) ? executor.parseGetModelResult() : null;
+
+			return mLastResult;
+
+		} catch (final IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private String getCommand(final File file) {
+		var command = "eld";
```

This only works, if `eld` is in the `PATH` of the system, should we include an Eldarica build in `releaseScripts`?

---

We need to make this a bit more portable, I agree.

- `eld` is a bash script, so it won't work on Windows. It sets some parameters and then runs a JAR -- we should implement that directly.
- Do we want to check in the eldarica JAR like we do for the SMT solver binaries, or wouldn't it be better to still have it as a Maven dependency?

I just thought about whether we could actually just call eldarica's `main` method from Java, but that might cause all kinds of unexpected behaviour (e.g. memory remaining allocated, re-invocations behaving incorrectly due to old cached stuff, coupling of required java versions), so... maybe not.

---

> Do we want to check in the eldarica JAR like we do for the SMT solver binaries, or wouldn't it be better to still have it as a Maven dependency?

Yes, the Maven dependency is much nicer of course. But I am not sure how you could actually call the binary in that case.
If it is not straightforward to include Eldarica in `releaseScripts`, we might really just assume that `eld` is in the PATH of the system for now (maybe you can throw an useful exception in that case?).
The best solution would be of course to get rid of `EldaricaCliChcScript` again and call Eldarica programmatically. If we cannot get it working, we could include the JAR along with our custom launchers (incl. for Windows) in `releaseScripts`. But this is more of a long term question 😉